python/mlx/nn/layers/base.py

```diff
@@ -439,3 +439,11 @@ def _set_train(_, m):
 
     def eval(self):
         self.train(False)
+
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        state.pop("_compiled_call_impl", None)
```

Comment:
What is `_compiled_call_impl` for ?

Reply:
I was confused whether to add this line or not. I think torch doesn't need this particular value from state dict while copying. I'll remove this line.

Reply:
nit, can just return the copy instead of creating the variable first

Reply:
done

Reply:
@dc-dc-dc pls review

Reply:
LGTM, nice addition 🚀